src/Compilers/Test/Utilities/CSharp/CSharpTestBase.cs

```diff
@@ -1050,6 +1050,21 @@ public static SyntaxTree ParseWithRoundTripCheck(string text, CSharpParseOptions
             string sourceFileName = "",
             bool skipUsesIsNullable = false) => CreateCompilationCore(source, TargetFrameworkUtil.GetReferences(TargetFramework.Mscorlib45, references), options, parseOptions, assemblyName, sourceFileName, skipUsesIsNullable, experimentalFeature: feature);
 
+        internal static CSharpCompilation CreateNumericIntPtrCompilation(
+              CSharpTestSource source,
+              IEnumerable<MetadataReference> references = null,
+              CSharpCompilationOptions options = null,
+              CSharpParseOptions parseOptions = null,
+              string assemblyName = "",
+              string sourceFileName = "")
+        {
+            // Note: we use skipUsesIsNullable and skipExtraValidation so that nobody pulls
+            // on the compilation or its references before we set the RuntimeSupportsNumericIntPtr flag.
+            var comp = CreateCompilationCore(source, references, options, parseOptions, assemblyName, sourceFileName, skipUsesIsNullable: true, experimentalFeature: null, skipExtraValidation: true);
+            comp.Assembly.RuntimeSupportsNumericIntPtr = true;
```

Comment:
You mentioned DIMs used a similar solution here for testing. I wondered why we didn't just declare the RuntimeFeature type ourselves with the members present that indicate the feature is supported? Do you know?

Reply:
RuntimeFeature is not just a well-known type, it's a special type. So it must be in corlib along with `object`. That makes it hard to inject into a compilation.